hydromt/raster.py

```diff
@@ -1175,6 +1175,97 @@ def rasterize(
         da_out.raster.set_attrs(**self.attrs)
         return da_out
 
+    def rasterize_geometry(
```

Comment:
Thanks for your comments! I can change the name of the function indeed to make it more explicit.
I'm not sure I fully understand your comment on ``vector_grid`` method. I guess for computing polygon area, what matters is the intersection with the grid so should be fine if grid is a polygon or line type. For fraction this is a little more delicate as I need to make sure fraction <1 so the way cell area is calculated should match calculating the intersected polygon area. Maybe ``raster.area_grid`` would still work without needing the polygons?
In any case seems like it would be nice to commit your changes to this branch so that I can test and update if my new ``rasterize_polygon_area`` method still work with your changes!

Reply:
I've pushed the improved vector_grid method to #351